src/extractor/guidance/turn_handler.cpp

```diff
@@ -9,19 +9,39 @@
 #include <utility>
 
 #include <boost/assert.hpp>
+#include <boost/optional.hpp>
 
 using EdgeData = osrm::util::NodeBasedDynamicGraph::EdgeData;
 using osrm::extractor::guidance::getTurnDirection;
 using osrm::util::angularDeviation;
 
-namespace osrm
-{
-namespace extractor
-{
-namespace guidance
-{
 namespace
 {
+
+using namespace osrm::extractor::guidance;
+// given two adjacent roads and `road1` being a candidate for a fork,
+// return false, if next road `road2` is also a fork candidate or
+// return true, if `road2` is not a suitable fork candidate and thus, `road1` the outermost fork
```

Comment:
adjacent == any specific order? I'd assume they are still in ccw order? Would be nice to specify this.